op-node/p2p/cli/load_config.go

```diff
@@ -0,0 +1,301 @@
+	addrs := strings.Split(ctx.GlobalString(flags.StaticPeers.Name), ",")
+	for i, addr := range addrs {
+		addr = strings.TrimSpace(addr)
+		if addr == "" {
+			continue // skip empty multi addrs
+		}
+		a, err := multiaddr.NewMultiaddr(addr)
+		if err != nil {
+			return fmt.Errorf("failed to parse multi addr of static peer %d (out of %d): %q err: %w", i, len(addrs), addr, err)
+		}
+		conf.StaticPeers = append(conf.StaticPeers, a)
+	}
+
+	for _, v := range strings.Split(ctx.GlobalString(flags.HostMux.Name), ",") {
+		v = strings.ToLower(strings.TrimSpace(v))
+		var mc lconf.MsMuxC
+		var err error
+		switch v {
+		case "yamux":
+			mc, err = p2p.YamuxC()
+		case "mplex":
+			mc, err = p2p.MplexC()
+		default:
+			return fmt.Errorf("could not recognize mux %s", v)
+		}
+		if err != nil {
+			return fmt.Errorf("failed to make %s constructor: %w", v, err)
+		}
+		conf.HostMux = append(conf.HostMux, mc)
+	}
+
+	secArr := strings.Split(ctx.GlobalString(flags.HostSecurity.Name), ",")
+	for _, v := range secArr {
+		v = strings.ToLower(strings.TrimSpace(v))
+		var sc lconf.MsSecC
+		var err error
+		switch v {
+		case "none": // no security, for debugging etc.
+			if len(conf.HostSecurity) > 0 || len(secArr) > 1 {
+				return errors.New("cannot mix secure transport protocols with no-security")
+			}
+			conf.NoTransportSecurity = true
+		case "noise":
+			sc, err = p2p.NoiseC()
+		case "tls":
+			sc, err = p2p.TlsC()
+		default:
+			return fmt.Errorf("could not recognize security %s", v)
+		}
+		if err != nil {
+			return fmt.Errorf("failed to make %s constructor: %w", v, err)
+		}
+		conf.HostSecurity = append(conf.HostSecurity, sc)
+	}
+
+	conf.PeersLo = ctx.GlobalUint(flags.PeersLo.Name)
+	conf.PeersHi = ctx.GlobalUint(flags.PeersHi.Name)
+	conf.PeersGrace = ctx.GlobalDuration(flags.PeersGrace.Name)
+	conf.NAT = ctx.GlobalBool(flags.NAT.Name)
+	conf.UserAgent = ctx.GlobalString(flags.UserAgent.Name)
+	conf.TimeoutNegotiation = ctx.GlobalDuration(flags.TimeoutNegotiation.Name)
+	conf.TimeoutAccept = ctx.GlobalDuration(flags.TimeoutAccept.Name)
+	conf.TimeoutDial = ctx.GlobalDuration(flags.TimeoutDial.Name)
+
+	peerstorePath := ctx.GlobalString(flags.PeerstorePath.Name)
+	if peerstorePath == "" {
+		return errors.New("peerstore path must be specified, use 'memory' to explicitly not persist peer records")
+	}
+
+	var err error
+	var store ds.Batching
+	if peerstorePath == "memory" {
+		store = sync.MutexWrap(ds.NewMapDatastore())
+	} else {
+		store, err = leveldb.NewDatastore(peerstorePath, nil) // default leveldb options are fine
+		if err != nil {
+			return fmt.Errorf("failed to open leveldb db for peerstore: %w", err)
+		}
+	}
+	conf.Store = store
+
+	return nil
+}
+
+func loadNetworkPrivKey(ctx *cli.Context) (*crypto.Secp256k1PrivateKey, error) {
+	raw := ctx.GlobalString(flags.P2PPrivRaw.Name)
+	if raw != "" {
+		return parsePriv(raw)
+	}
+	keyPath := ctx.GlobalString(flags.P2PPrivPath.Name)
```

Comment:
Default keyPath if its empty string? Other places in the config parsing handle empty string. The defaults could all be handled when defining the flags, but also doesn't prevent somebody from passing the empty string so prob best to check for it here

Reply:
This code is not part of the changes either, but agree we could change it. The flag definition defines the default as non-empty. If someone changes it explicitly to be empty, it's likely a configuration error, so returning an error is maybe the right behavior.

Reply:
wouldn't be opposed to erroring here if its empty even tho its a diff on top of moving the code

Reply:
opened a separate PR for it, let's get this merged first: https://github.com/ethereum-optimism/optimism/pull/4676